src/core/crypto_aos.c

```diff
@@ -359,21 +359,31 @@ int32_t Crypto_AOS_ApplySecurity(uint8_t *pTfBuffer)
 
     // Get Key
     crypto_key_t *ekp = NULL;
+    crypto_key_t *akp = NULL;
     ekp               = key_if->get_key(sa_ptr->ekid);
-    if (ekp == NULL)
+    akp               = key_if->get_key(sa_ptr->akid);
+
+    if (ekp->key_state != KEY_ACTIVE || akp->key_state != KEY_ACTIVE)
```

Comment:
Line 366: Should this be an AND? As in, if you get this far and neither key is active, then there's a problem.

Does this introduce a segfault by not first checking if ekp == null, or akp==null before accessing the key_state?